apps/frontend/src/libs/components/router-provider/protected-route.tsx

```diff
@@ -0,0 +1,26 @@
+import { FC, ReactNode } from "react";
+
+import { Navigate } from "~/libs/components/components.js";
+import { AppRoute } from "~/libs/enums/enums.js";
+import { useAppSelector, useLocation } from "~/libs/hooks/hooks.js";
+import { ValueOf } from "~/libs/types/types.js";
+
+type Properties = {
+	children: ReactNode;
+	redirectTo: ValueOf<typeof AppRoute>;
+};
+
+const ProtectedRoute: FC<Properties> = ({ children, redirectTo }) => {
```

Comment:
```suggestion
type Properties = {
	children: React.ReactNode;
	redirectTo: ValueOf<typeof AppRoute>;
};

const ProtectedRoute: React.FC<Properties> = ({ children, redirectTo }) => {
```
Let's use the same approach everywhere